salt/modules/kubernetesmod.py

```diff
@@ -1569,7 +1569,7 @@ def __dict_to_service_spec(spec):
         if key == 'ports':
             spec_obj.ports = []
             for port in value:
-                kube_port = kubernetes.client.V1ServicePort()
+                kube_port = kubernetes.client.V1ServicePort(port=port.get('port', ''))
```

Comment:
does this only work with kubernetes >=4.0.0 now? if so is it possible to keep compatibility for both 3 and 4 kubernetes versions?

Reply:
A good question, thank you! I did just verify now that this still works for kubernetes-python below 4.0.0 as well.